src/ringmesh/io/geomodel_builder_resqml.cpp

```diff
@@ -0,0 +1,968 @@
+                        layer3d.name() );
+
+                    builder_.geology.add_parent_children_relation(
+                        layer, region.gmme() );
+                }
+
+                // build the stati column
+                RockFeature rock( "rock", ROCKTYPE::NONE );
+                std::vector< std::shared_ptr< const StratigraphicUnit > > units;
+
+                for( auto unit_key :
+                    rank_interp->getStratigraphicUnitInterpretationSet() )
+                {
+                    const UnitInfo& unit = unit_2_info_[unit_key];
+                    const Interface3D* top =
+                        unit.interface_top_.is_defined()
+                            ? dynamic_cast< const Interface3D* >(
+                                  &geomodel_.geological_entity(
+                                      Interface3D::type_name_static(),
+                                      unit.interface_top_.index() ) )
+                            : nullptr;
+                    const Interface3D* base =
+                        unit.interface_base_.is_defined()
+                            ? dynamic_cast< const Interface3D* >(
+                                  &geomodel_.geological_entity(
+                                      Interface3D::type_name_static(),
+                                      unit.interface_base_.index() ) )
+                            : nullptr;
+
+                    std::shared_ptr< const StratigraphicUnit > sunit(
+                        new UnsubdividedStratigraphicUnit( unit.name_, base,
+                            top,
+                            static_cast< const Layer3D& >(
+                                geomodel_.geological_entity(
+                                    Layer3D::type_name_static(),
+                                    unit.layer_.index() ) ),
+                            unit.relation_top_, unit.relation_base_, rock, 0,
+                            10 ) );
+
+                    units.push_back( sunit );
+
+#ifdef RINGMESH_DEBUG
+                    std::cout << " relation base: " << (int) unit.relation_base_
+                              << std::endl;
+                    std::cout << " relation top: " << (int) unit.relation_top_
+                              << std::endl;
+                    std::cout
+                        << " interface base: " << unit.interface_base_.index()
+                        << std::endl;
+                    std::cout
+                        << " interface top: " << unit.interface_top_.index()
+                        << std::endl;
+#endif
+                }
+
+                StratigraphicColumn* strati =
+                    new StratigraphicColumn( column->getTitle(), units,
+                        STRATIGRAPHIC_PARADIGM::CHRONOSTRATIGRAPHIC );
+                geomodel_.set_stratigraphic_column( strati );
+            }
+        }
+
+        if( stratiColumnSet.empty() )
+        {
+            if( !geomodel_.entity_type_manager()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1066)